src/amqp-client/connection.cr

```diff
@@ -225,13 +226,13 @@ class AMQP::Client
       io.read_timeout = nil
     end
 
-    private def self.start(io, user, password, name)
+    private def self.start(io, user, password, name, client_args)
       io.write AMQ::Protocol::PROTOCOL_START_0_9_1.to_slice
       io.flush
       Frame.from_io(io) { |f| f.as?(Frame::Connection::Start) || raise Error::UnexpectedFrame.new(f) }
       props = Arguments.new({
         connection_name: name,
-        product:         "amqp-client.cr",
+        product:         client_args || "amqawegrdp-client.cr",
```

Comment:
`client_args` is not a good name, that would be the whole `Argument` structure.

```suggestion
        product:         product_name || "amqp-client.cr",
```

But if you specify the product name, you probably would like to be able to set the product `version`, and maybe the `platform` as well. Which makes this PR a whole lot more complex, adding 3 arguments to all those methods..

Reply:
Yes, i started down that road (hence the name) but it got messy.. doesn't really get better with something like a named tuple either, almost more clean to just add more arguments